client/src/components/client-area/ProgressOrder.vue

```diff
@@ -0,0 +1,313 @@
+<!-- Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS -->
+
+<template>
+  <template v-if="!querying && !error && customOrderBmsStatus !== CustomOrderStatus.Unknown && customOrderSellsyStatus">
+    <div class="process-container">
+      <ms-report-text
+        v-if="customOrderSellsyStatus === CustomOrderRequestStatus.Standby"
+        :theme="MsReportTheme.Warning"
+        class="process-stop-container"
+      >
+        <div class="process-stop-text">
+          <ion-text class="title-h4">{{ $msTranslate('clientArea.dashboard.step.standby.title') }}</ion-text>
+          <ion-text class="body">{{ $msTranslate('clientArea.dashboard.step.standby.description') }}</ion-text>
+        </div>
+      </ms-report-text>
+
+      <ms-report-text
+        v-if="customOrderSellsyStatus === CustomOrderRequestStatus.Cancelled"
+        :theme="MsReportTheme.Error"
+        class="process-stop-container"
+      >
+        <div class="process-stop-text">
+          <ion-text class="title-h4">{{ $msTranslate('clientArea.dashboard.step.cancel.title') }}</ion-text>
+          <ion-text class="body">{{ $msTranslate('clientArea.dashboard.step.cancel.description') }}</ion-text>
+        </div>
+      </ms-report-text>
+      <div
+        class="process-step"
+        v-for="(step, index) in steps"
+        :key="index"
+        :class="{
+          'process-step-todo': getStep(customOrderBmsStatus, customOrderSellsyStatus) < getStep(step.statusBms, step.statusRequest),
+          'process-step-active': getStep(customOrderBmsStatus, customOrderSellsyStatus) === getStep(step.statusBms, step.statusRequest),
+          'process-step-done': getStep(customOrderBmsStatus, customOrderSellsyStatus) > getStep(step.statusBms, step.statusRequest),
+        }"
+      >
+        <div class="process-step-icon">
+          <div
+            class="dot-todo"
+            v-if="getStep(customOrderBmsStatus, customOrderSellsyStatus) < getStep(step.statusBms, step.statusRequest)"
+          />
+          <div
+            class="dot-active"
+            v-if="getStep(customOrderBmsStatus, customOrderSellsyStatus) === getStep(step.statusBms, step.statusRequest)"
+          />
+          <ion-icon
+            class="process-step-icon__item"
+            :icon="checkmarkCircle"
+            v-if="getStep(customOrderBmsStatus, customOrderSellsyStatus) > getStep(step.statusBms, step.statusRequest)"
```

Comment:
The logic confuses me:
You create an array of object that you use to get a key of an object to get an index. Can't you just create your initial array in the right order and just compare with the variable `index`? And just have a variable to store the `getStep(customOrderBmsStatus, customOrderSellsyStatus)` directly when retrieving them. Which updates all this to
`v-if="customOrderIndex > index"`